tests/test-grad0.c

```diff
@@ -215,15 +215,35 @@ bool check_gradient(
     }
 
     struct ggml_cgraph gf = ggml_build_forward (f);
-    gf.n_threads = n_threads;
 
     struct ggml_cgraph gb = ggml_build_backward(ctx0, &gf, false);
```

Comment:
Sadly, ggml_build_backward was broken, I'll dig into it.

Reply:
Clean build by deleting `build` dir  resolved this issue, no idea what's wrong. The output is similar to master branch.

```
$ time ./bin/test-grad0 
test-grad0: iter:0/4
GGML_N_THREADS = 4
test-grad0: iter:1/4
test-grad0: iter:2/4
cross_entropy_loss: ndims=3, i=0, k=0, x0=-0.951187, xm=-1.051187, xp=-0.851187, f0=11.762405, f1=11.762405, g0=0.000000, g1=-0.571868, eps=0.100000, error_abs=0.571868, error_rel=0.000000
test-grad0: iter:3/4
```